File: system-test/java/src/main/java/com/optivem/shop/testkit/dsl/port/given/steps/GivenClock.java
package com.optivem.shop.testkit.dsl.port.given.steps;

import com.optivem.shop.testkit.dsl.port.given.steps.base.GivenStep;

public interface GivenClock extends GivenStep {
    GivenClock withTime();
    GivenClock withTime(String time);
    GivenClock withWeekday();
    GivenClock withWeekend();
}

File: system-test/typescript/src/testkit/dsl/port/given/steps/given-clock.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenClock {
  withTime(time?: string): this;
  withWeekday(): this;
  withWeekend(): this;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
